web/src/main/java/com/epam/esm/web/exceptionhandler/ControllerExceptionHandler.java

```diff
@@ -8,13 +8,32 @@
 import com.epam.esm.web.dto.ExceptionDTO;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.context.request.WebRequest;
 
+import javax.naming.AuthenticationException;
+
 @ControllerAdvice
 public class ControllerExceptionHandler {
 
+    @ExceptionHandler({AuthenticationException.class})
```

Comment:
Use filters and existing functionality https://docs.spring.io/spring-security/site/docs/4.2.20.RELEASE/apidocs/org/springframework/security/web/access/ExceptionTranslationFilter.html

Reply:
I have AuthenticationEntryPoint and AccessDeniedHandler

Reply:
Need to discuss